modules/apps/object/object-service/src/main/java/com/liferay/object/internal/related/models/SystemObject1toMObjectRelatedModelsProviderImpl.java

```diff
@@ -333,6 +333,83 @@ public List<T> getUnrelatedModels(
 						return groupIdColumn.eq(groupId);
 					}
 				).and(
+					() -> {
+						Column<?, Long> primaryKeyColumn = _table.getColumn(
+							objectDefinition.getPKObjectFieldDBColumnName());
+
+						DynamicObjectDefinitionTable
+							dynamicObjectDefinitionTable =
+								_getDynamicObjectDefinitionTable();
+						ObjectField objectField =
+							_objectFieldLocalService.getObjectField(
+								objectRelationship.getObjectFieldId2());
+
+						Column<DynamicObjectDefinitionTable, Long>
+							foreignKeyColumn =
+								(Column<DynamicObjectDefinitionTable, Long>)
+									dynamicObjectDefinitionTable.getColumn(
+										objectField.getDBColumnName());
+
+						return primaryKeyColumn.notIn(
+							DSLQueryFactoryUtil.select(
+								dynamicObjectDefinitionTable.
+									getPrimaryKeyColumn()
+							).from(
+								dynamicObjectDefinitionTable
+							).where(
+								foreignKeyColumn.neq(0L)
+							).limit(
+								start, end
+							));
+					}
+				)
+			));
+	}
+
+	@Override
+	public int getUnrelatedModelsCount(
+			long companyId, long groupId, ObjectDefinition objectDefinition,
+			long objectEntryId, long objectRelationshipId)
+		throws PortalException {
+
+		Column<?, Long> companyIdColumn = (Column<?, Long>)_table.getColumn(
+			"companyId");
+
+		ObjectRelationship objectRelationship =
+			_objectRelationshipLocalService.getObjectRelationship(
+				objectRelationshipId);
+
+		ObjectDefinition objectDefinition1 =
+			_objectDefinitionLocalService.getObjectDefinition(
+				objectRelationship.getObjectDefinitionId1());
+
+		PersistedModelLocalService persistedModelLocalService =
+			_persistedModelLocalServiceRegistry.getPersistedModelLocalService(
+				objectDefinition.getClassName());
+
+		return persistedModelLocalService.dslQueryCount(
+			DSLQueryFactoryUtil.count(
+			).from(
+				_table
+			).where(
+				companyIdColumn.eq(
+					companyId
+				).and(
+					() -> {
+						Column<?, Long> groupIdColumn = _table.getColumn(
+							"groupId");
+
+						if ((groupIdColumn == null) ||
+							Objects.equals(
+								ObjectDefinitionConstants.SCOPE_COMPANY,
+								objectDefinition1.getScope())) {
+
+							return null;
+						}
+
+						return groupIdColumn.eq(groupId);
```

Comment:
If it is scoped by company, then will always return null, otherwise use the `groupIdColumn`. Am I right?
```suggestion
						if (Objects.equals(
								ObjectDefinitionConstants.SCOPE_COMPANY,
								objectDefinition1.getScope())) {

							return null;
						}
						
						Column<?, Long> groupIdColumn = _table.getColumn(
							"groupId");

						return groupIdColumn.eq(groupId);
```